light/client.go

```diff
@@ -24,9 +24,23 @@ const (
 	skipping
 
 	defaultPruningSize = 1000
-	// For verifySkipping, when using the cache of headers from the previous batch,
-	// they will always be at a height greater than 1/2 (normal verifySkipping) so to
-	// find something in between the range, 9/16 is used.
+
+	// For verifySkipping, we need an algorithm to find what height to check
+	// next to see if it has sufficient validator set overlap. The most
+	// intuitive method is to take the halfway point i.e. if you trusted block
+	// 1 and were not able to verify block 128 then your next try would be 64.
+	//
+	// However this implementation caches all the prior results which means it
+	// isn't optimal to take the halfway point. Take this simple example. Say
```

Comment:
Ho
```suggestion
	// However, because this implementation caches all the prior results, instead of always taking halfpoints
	// it is more efficient to re-check cached blocks. Take this simple example. Say
```